crates/typst-library/src/layout/par.rs

```diff
@@ -736,6 +740,55 @@ fn prepare<'a>(
     })
 }
 
+/// Add some spacing between Han characters and western characters.
+/// See Requirements for Chinese Text Layout, Section 3.2.2 Mixed Text Composition in Horizontal
+/// Written Mode
+fn add_cjk_latin_spacing(items: &mut [Item]) {
+    let mut items = items.iter_mut().peekable();
+    let mut prev: Option<&ShapedGlyph> = None;
+    while let Some(item) = items.next() {
+        if let Item::Text(ref mut text) = item {
```

Comment:
We can get rid of some indentation here with `let .. else ..`. I would also say that `if let Item::Text(text) = &mut item` is a bit more idiomatic nowadays.

Reply:
Fixed in 2aa1bce.